tests/test_env_path_safety.py

```
@@ -0,0 +1,54 @@
+        # tmp_path is not under HOME in all environments, but pytest's
+        # tmp_path is under tmpdir. For this test, point at a real
+        # HOME-relative location.
+        target = Path.home() / "_pytest_env_path_target"
+        try:
+            monkeypatch.setenv("SIEGE_TEST_VAR", str(target))
+            result = _env_path("SIEGE_TEST_VAR", Path.home() / "fallback")
+            assert result == target.resolve()
+        finally:
+            pass  # nothing to clean — target was never created
+
+    def test_outside_home_rejected_by_default(self, monkeypatch, caplog):
```

Ruff (0.15.12)

[warning] Unused method argument: `caplog`

(ARG002)